text/0117-typo-setting-api.md

```diff
@@ -0,0 +1,295 @@
+
+- 🔴 When Meilisearch is secured, if the API Key do not have the `indexes.create` action defined, the API returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error in the related asynchronous `task` resource. See [3.2.2.2.1. Response Definition](#32221-response-definition).
+
+> Otherwise, Meilisearch will create the index in a lazy way. See [3.2.2.2.3. Lazy Index Creation](#32223-lazy-index-creation).
+
+- 🔴 Sending invalid integer values for the `minWordSizeForTypos` object fields returns an [invalid_typo_min_word_size_for_typos](0061-error-format-and-definitions.md#invalid_typo_min_word_size_for_typos) error.
+
+##### 3.2.2.2.3. Lazy Index Creation
+
+If the requested `index_uid` does not exist, and the authorization layer allows it (See [3.2.2.2.2.1. Async Errors](#322221-async-errors)), Meilisearch will create the index when the related asynchronous task resource is executed. See [3.2.2.2.1. Response Definition](#32221-response-definition).
+
+#### 3.2.2.3. `DELETE`- `indexes/:index_uid/settings/typo`
+
+Allow resetting the typo tolerance feature to the default for an index.
+
+##### 3.2.2.3.1. Response Definition
+
+When the request is in a successful state, Meilisearch returns the HTTP code `202 Accepted`. The response's content is the summarized representation of the received asynchronous task.
+
+See [Summarized `task` Object for `202 Accepted`](0060-tasks-api.md#summarized-task-object-for-202-accepted).
+
+##### 3.2.2.3.2. Errors
+
+###### 3.2.2.3.2.1. Asynchronous Index Not Found Error
+
+- 🔴 If the requested `index_uid` does not exist, the API returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error in the related async `task` resource. See [3.2.2.3.1. Response Definition](#32231-response-definition).
+
+### 3.2.3. General Errors
+
+These errors apply to all endpoints described here.
+
+#### 3.2.3.1. Auth Errors
+
+The auth layer can return the following errors if Meilisearch is secured (a master-key is defined).
+
+- 🔴 Accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 Accessing this route with a key that does not have permissions (i.e. other than the master-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+## 2. Technical Details
+
+### 2.1. Typos Calculation
+
+The size of a query term triggers the typo tolerance feature.
+
+By default, a query term made of `5` characters can tolerate `1` typo, while a query term made of `9` characters can tolerate `2` typos.
+
+If a query term can be derived from a distance of one or two typos to be found in a document, it will be selected as a candidate. The number of typos found has an impact on the relevancy.
+
+#### 2.1.1 Typo On The First Character
+
+A query term having the first character considered as a typo will not count as `1` typo but as `2` typos.
```

Comment:
- Just to be clear: if I search for `qorld` instead of `world`, the `q` will be two typos? 
- How will this work if, e.g., my settings only allow for 1 typo per 5 characters, will I not get any results?

Reply:
> 1. Just to be clear: if I search for `qorld` instead of `world`, the `q` will be two typos?

Indeed, the engine will only return documents that contains `world` after documents containing `qoyld` for example (1 typo).

> 2. How will this work if, e.g., my settings only allow for 1 typo per 5 characters, will I not get any results?

Exactly, the engine will not try to find worlds with 2 typos on a word that is smaller than 5 letters, a typo on the first letter is considered 2 therefore you will never find documents with `world` for example.